Detectors/EMCAL/base/include/EMCALBase/Point.h

```diff
@@ -83,10 +90,11 @@ namespace AliceO2 {
       void PrintStream(std::ostream &stream) const;
       
     private:
-      Int_t         mShunt;             ///< Shunt (check if needed)
-      Int_t         mPrimary;           ///< Primary particles at the origin of the hit
-      Int_t         mParent;            ///< Parent particle that entered the EMCAL
-      Double32_t    mInitialEnergy;     ///< Energy of the parent particle that entered the EMCAL
+      Vector3D<float>   mPvector;           ///< Momentum Vector
+      Int_t             mShunt;             ///< Shunt (check if needed)
```

Comment:
it seems you are keeping the momentum in float mode. It would then also make sense to adapt your constructor to go away from `Double_t *pos` + `Double_t *mom` to `Point3D<float> pos` + `Vector3D<float> mom`. We should not use pointer types anymore ( they have various problems such as aliasing).